plugins/cost-insights/src/components/EntityCosts/EntityCosts.tsx

```diff
@@ -0,0 +1,161 @@
+import { stringifyEntityRef } from '@backstage/catalog-model';
+
+export const EntityCostsCard = () => {
+  const client = useApi(costInsightsApiRef);
+  const { entity } = useEntity();
+
+  const {
+    loadingActions,
+    loadingGroups,
+    loadingBillingDate,
+    loadingInitial,
+    dispatchInitial,
+    dispatchInsights,
+    dispatchNone,
+  } = useLoading(mapLoadingToProps);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  // The dispatchLoading functions are derived from loading state using mapLoadingToProps, to
+  // provide nicer props for the component. These are re-derived whenever loading state changes,
+  // which causes an infinite loop as product panels load and re-trigger the useEffect below.
+  // Since the functions don't change, we can memoize - but we trigger the same loop if we satisfy
+  // exhaustive-deps by including the function itself in dependencies.
+
+  const dispatchLoadingInitial = useCallback(dispatchInitial, []);
+  const dispatchLoadingInsights = useCallback(dispatchInsights, []);
+  const dispatchLoadingNone = useCallback(dispatchNone, []);
+  /* eslint-enable react-hooks/exhaustive-deps */
+
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const [dailyCost, setDailyCost] = useState<Maybe<Cost>>(null);
+  const [error, setError] = useState<Maybe<Error>>(null);
+  const { pageFilters } = useFilters(p => p);
+
+  useEffect(() => {
+    async function getInsights() {
+      setError(null);
+      try {
+        dispatchLoadingInsights(true);
+        const intervals = intervalsOf(
+          pageFilters.duration,
+          lastCompleteBillingDate,
+        );
+
+        const fetchedDailyCost = await client.getCatalogEntityDailyCost!(
+          stringifyEntityRef(entity),
+          intervals,
+        );
+        setDailyCost(fetchedDailyCost);
+      } catch (e) {
+        setError(e);
+        dispatchLoadingNone(loadingActions);
+      } finally {
+        dispatchLoadingNone(loadingActions);
+        dispatchLoadingInitial(false);
+        dispatchLoadingInsights(false);
+      }
+    }
+
+    // Wait for metadata to finish loading
+    if (!loadingBillingDate) {
+      getInsights();
+    }
+  }, [
+    client,
+    entity,
+    pageFilters,
+    loadingActions,
+    loadingGroups,
+    loadingBillingDate,
+    dispatchLoadingInsights,
+    dispatchLoadingInitial,
+    dispatchLoadingNone,
+    lastCompleteBillingDate,
+  ]);
+
+  if (loadingInitial) {
+    return <Progress />;
+  }
+
+  if (error) {
+    return <MaterialAlert severity="error">{error.message}</MaterialAlert>;
+  }
+
+  if (!dailyCost) {
+    return <MaterialAlert severity="error">No daily costs</MaterialAlert>;
+  }
+
+  return <CostOverviewCard dailyCostData={dailyCost} metricData={null} />;
+};
+
+export const EntityCosts = () => {
+  const client = useApi(costInsightsApiRef);
+
+  if (!client.getCatalogEntityDailyCost) {
+    return (
+      <WarningPanel
```

Comment:
I'm assuming we want to have this error here because you "opt in" by deliberately adding the `<EntityCosts />` component to your page right? :) Otherwise it might be a bit intrusive and perhaps worth skipping rendering entirely or possibly log instead.

Reply:
That was the idea, yeah. I thought it was an easy way to give feedback to someone in the process of setting up cost insights in their Backstage instance. I have no strong preference though, if you'd rather skip rendering + log somewhere instead.

Reply:
I think it make sense to have it for someone that want to start using this new feature :) It would have been another thing if it was displaying on for example the EntityPage where everyone suddenly had to implement the API to not have the error but it makes sense for a completely new page like this.